File: Core/src/main/kotlin/jokatu/game/joining/JoinInputDeserialiser.java
package jokatu.game.joining;

import jokatu.game.input.DeserialisationException;
import jokatu.game.input.SingleKeyInputDeserialiser;
import org.jetbrains.annotations.NotNull;
import org.springframework.stereotype.Component;

import java.util.Map;

/**
 * Deserialises {@link JoinInput}s.  The correct JSON should look like <code>{"join": true}</code>.
 */
@Component
public class JoinInputDeserialiser extends SingleKeyInputDeserialiser<JoinInput> {
	@NotNull
	@Override
	public JoinInput deserialiseSingleValue(@NotNull Map<String, Object> json, @NotNull Object value) throws DeserialisationException {
		return new JoinInput();
	}

	@NotNull
	@Override
	protected String getKeyName() {
		return "join";
	}
}

File: Core/src/main/kotlin/jokatu/game/joining/JoinInputDeserialiser.kt
package jokatu.game.joining

import jokatu.game.input.DeserialisationException
import jokatu.game.input.SingleKeyInputDeserialiser
import org.springframework.stereotype.Component

/**
 * Deserialises [JoinInput]s.  The correct JSON should look like `{"join": true}`.
 */
@Component
class JoinInputDeserialiser : SingleKeyInputDeserialiser<JoinInput>() {

	override val keyName: String
		get() = "join"

	@Throws(DeserialisationException::class)
	override fun deserialiseSingleValue(json: Map<String, Any>, value: Any): JoinInput {
		return JoinInput()
	}
}
